src/appleseed-max-impl/appleseedplasticmtl/appleseedplasticmtl.cpp

```diff
@@ -0,0 +1,904 @@
+        .insert("Scattering", fmt_osl_expr(m_scattering / 100.0f))
+        .insert("IOR", fmt_osl_expr(m_ior)));
+
+    assembly.shader_groups().insert(shader_group);
+
+    //
+    // Material.
+    //
+
+    asr::ParamArray material_params;
+    material_params.insert("osl_surface", shader_group_name);
+
+    const std::string instance_name =
+        insert_texture_and_instance(
+            assembly,
+            m_alpha_texmap,
+            false,
+            asr::ParamArray(),
+            asr::ParamArray()
+                .insert("alpha_mode", "detect"));
+    if (!instance_name.empty())
+        material_params.insert("alpha_map", instance_name);
+    else material_params.insert("alpha_map", m_alpha / 100.0f);
+
+    return asr::OSLMaterialFactory().create(name, material_params);
+}
+
+asf::auto_release_ptr<asr::Material> AppleseedPlasticMtl::create_builtin_material(
+    asr::Assembly&  assembly,
+    const char*     name)
+{
+    asr::ParamArray material_params;
+    std::string instance_name;
+    const bool use_max_procedural_maps = true;
+
+    //
+    // BRDF.
+    //
+    {
+        asr::ParamArray bsdf_params;
+        bsdf_params.insert("mdf", "ggx");
+        bsdf_params.insert("highlight_falloff", m_highlight_falloff / 100.0f);
+        bsdf_params.insert("ior", m_ior);
+        bsdf_params.insert("internal_scattering", m_scattering / 100.0f);
+        bsdf_params.insert("specular_reflectance_multiplier", m_specular_weight / 100.0f);
+        bsdf_params.insert("diffuse_reflectance_multiplier", m_diffuse_weight / 100.0f);
+
+        // Specular.
+        instance_name = insert_texture_and_instance(assembly, m_specular_texmap, use_max_procedural_maps);
+        if (!instance_name.empty())
+            bsdf_params.insert("specular_reflectance", instance_name);
+        else
+        {
+            const auto color_name = std::string(name) + "_bsdf_normal_reflectance_color";
+            insert_color(assembly, m_specular, color_name.c_str());
+            bsdf_params.insert("specular_reflectance", color_name);
+        }
+
+        // Diffuse.
+        instance_name = insert_texture_and_instance(assembly, m_diffuse_texmap, use_max_procedural_maps);
+        if (!instance_name.empty())
+            bsdf_params.insert("diffuse_reflectance", instance_name);
+        else
+        {
+            const auto color_name = std::string(name) + "_bsdf_edge_tint";
+            insert_color(assembly, m_diffuse, color_name.c_str());
+            bsdf_params.insert("diffuse_reflectance", color_name);
+        }
+
+        // Roughness.
+        instance_name = insert_texture_and_instance(assembly, m_roughness_texmap, use_max_procedural_maps);
+        if (!instance_name.empty())
+            bsdf_params.insert("roughness", instance_name);
+        else bsdf_params.insert("roughness", m_roughness / 100.0f);
+
+        // BRDF.
+        const auto bsdf_name = std::string(name) + "_bsdf";
```

Comment:
Same remark here, `_brdf` would make more sense (same thing for the names of the linked entities).